main.py

```diff
@@ -0,0 +1,96 @@
+        # Check for and remove any empty strings in the input list
+        name_list = check_for_empty_string(name_list)
+        break
+    return name_list
+
+def get_valid_num_of_names(name_list):
+    """Get a user input for the number that is an integer and lower than the number of names in the list submitted."""
+    while True:
+        # Ask the user to input a number lower than the number of names in the list
+        num_of_names = input(f"Enter a number {len(name_list)} or lower to randomly select that number of names from the list: ")
+        # Check that the input is a valid integer
+        if not num_of_names.isdigit():
+            print("Error: Number must be an integer.")
+        # Check that the input number is less than or equal to the number of names in the list
+        elif int(num_of_names) > len(name_list):
+            print(f"Error: Number must be {len(name_list)} or lower.")
+        # Check that the input number is greater than 0
+        elif int(num_of_names) <= 0:
+            print(f"Error: Number must be greater than 0")
+        else:
+            return int(num_of_names)
+
+def check_punctuation(name_list):
+    """Check if name list contains only letters, hyphens, and commas."""
+    # Check if the input contains any characters other than letters, hyphens, commas, or whitespace
+    if re.search(r"[^a-zA-Z,\-\s]", name_list):
+        return False
+    return True
+
+def remove_duplicates(name_list):
+    """Remove duplicates from the name list."""
+    # Convert the input list to a set to remove duplicates, then convert back to a list
+    return list(set(name.lower() for name in name_list))
+
+def remove_whitespace_after_commas(name_list):
+    """Remove whitespace after commas in the name list."""
+    # Replace any whitespace that comes after a comma with just a comma
+    return re.sub(r',\s*', ',', name_list)
+
+def check_name_list_not_empty(name_list):
+    """Check if the name list is not empty."""
+    # Check if the input list is empty
+    if not name_list:
+        return False
+    return True
+
+def check_for_empty_string(name_list):
+    """Check for empty strings in a list and remove them."""
+    # Use a list comprehension to remove any empty strings from the input list
+    name_list = [name for name in name_list if name != '']
+    return name_list
+
+def select_random_names():
+    """Generate then print randomly selected names from the list of names submitted by the user"""
+    # Get the input list of names
+    name_list = get_user_inputs()
+    if len(name_list) == 1:
+        print(f"You only entered 1 name: {name_list[0]}")
+    # Get the number of names to select
+    num_of_names = get_valid_num_of_names(name_list)
+    # Create empty set to add random names to
+    selected_names = set()
+    # While the length of selected names is less than the number of names, add a randomly chosen name from the list
+    while len(selected_names) < num_of_names:
```

Comment:
Or you can use k parameter (An integer defining the length of the returned list), without using while loop and set

Reply:
It's a bit cumbersome but I did this to make sure no duplicates names would end up in the final list